python_modules/libraries/dagster-k8s/dagster_k8s/job.py

```diff
@@ -270,106 +270,65 @@ def __new__(
+                    is_required=True,
+                    description="The ``name`` of an existing Volume to mount into the pod in order to "
+                    "provide a ConfigMap for the Dagster instance. This Volume should contain a "
+                    "``dagster.yaml`` with appropriate values for run storage, event log storage, etc.",
                 ),
-                is_required=False,
-                default_value=[],
-                description="A list of volume mounts to include in the job's container. Default: ``[]``. See: "
-                "https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#volumemount-v1-core",
-            ),
-            "volumes": Field(
-                Array(
-                    Permissive(
-                        {
-                            "name": str,
-                        }
-                    )
+                "postgres_password_secret": Field(
+                    StringSource,
+                    is_required=False,
+                    description="The name of the Kubernetes Secret where the postgres password can be "
+                    "retrieved. Will be mounted and supplied as an environment variable to the Job Pod."
+                    'Secret must contain the key ``"postgresql-password"`` which will be exposed in '
+                    "the Job environment as the environment variable ``DAGSTER_PG_PASSWORD``.",
                 ),
-                is_required=False,
-                default_value=[],
-                description="A list of volumes to include in the Job's Pod. Default: ``[]``. For the many "
-                "possible volume source types that can be included, see: "
-                "https://v1-18.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#volume-v1-core",
-            ),
-        }
+                "dagster_home": Field(
+                    StringSource,
+                    is_required=False,
+                    default_value=DAGSTER_HOME_DEFAULT,
+                    description="The location of DAGSTER_HOME in the Job container; this is where the "
+                    "``dagster.yaml`` file will be mounted from the instance ConfigMap specified here. "
+                    "Defaults to /opt/dagster/dagster_home.",
+                ),
+                "load_incluster_config": Field(bool, is_required=False, default_value=True),
+                "kubeconfig_file": Field(Noneable(str), is_required=False, default_value=None),
+            },
+            DagsterK8sJobConfig.config_type_job(),
+        )
 
     @classmethod
-    def config_type_pipeline_run(cls, default_image_pull_policy=None):
+    def config_type_executor(cls):
+        return merge_dicts(
+            {
+                "job_image": Field(
+                    Noneable(StringSource),
+                    is_required=False,
+                    description="Docker image to use for launched task Jobs. If the repository is not "
```

Comment:
idk what a launched task Job is haha. Maybe just `image to use for launched Jobs`